src/view.rs

```diff
@@ -859,3 +859,87 @@ mod view_response_builder_tests {
         assert_eq!(expected, got);
     }
 }
+
+pub struct AllDocumentsViewValue {
+    rev: Revision,
+}
+
+impl AllDocumentsViewValue {
+    pub fn revision(&self) -> &Revision {
+        &self.rev
+    }
+}
+
+impl serde::Serialize for AllDocumentsViewValue {
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error>
+        where S: serde::Serializer
+    {
+        Ok(try!(serializer.serialize_struct_elt("rev", &self.rev)))
+    }
+}
+
+impl serde::Deserialize for AllDocumentsViewValue {
```

Comment:
The alternative is to use [serde_codegen](http://erickt.github.io/blog/2015/05/18/serde-0-dot-4-0-now-supports-macros-in-stable-rust/), which introduces its own complexity. For Chill, I prefer to hand-roll serialization and deserialization because:

1. The CouchDB API is stable, so,
2. Once you write the code for implementing `Serialize` or `Deserialize`, you're done, and the code will rarely change, if ever, and,
3. The code is mostly copy-and-paste anyway so it's not like it's hard to do.

Whereas, if we needed to continually change our JSON representations because, say, the HTTP API were continually changing, then the advantage would shift in favor of using serde_codegen and we would eat the extra build-time complexity.

Someday, when compiler plug-ins are stabilized in Rust, it will be a no-brainer to rip out the hand-rolled code and use serde macros.